plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Utils.hs

```diff
@@ -111,27 +112,35 @@ an under-approximation of how many arguments the term may need.
 e.g. consider the term @let id = \x -> x in id@: the variable @id@ has syntactic
 arity @[]@, but does in fact need an argument before it does any work.
 -}
-type Arity tyname name = [Param tyname name]
+type Arity tyname name uni ann = [Param tyname name uni ann]
 
 -- | Info attached to a let-binding needed for call site inlining.
 data VarInfo tyname name uni fun ann = MkVarInfo
     { varStrictness :: Strictness
     , varRhs        :: Term tyname name uni fun ann
     -- ^ its definition that has been unconditionally inlined.
-    , varArity      :: Arity tyname name
+    , varArity      :: Arity tyname name uni ann
     -- ^ its arity, storing to avoid repeated calculations.
     , varRhsBody    :: InlineTerm tyname name uni fun ann
     -- ^ the body of the function, for checking `acceptable` or not. Storing this to avoid repeated
     -- calculations.
     }
 -- | Is the next argument a term or a type?
-data Param tyname name =
-    TermParam name | TypeParam tyname
+data Param tyname name uni ann =
+    TermParam ann name (Type tyname uni ann)
```

Comment:
why put the extra info in? It's not used, and it's confusing what the point of the arity is, I think?